wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/monitoring/ClusterTopologyMonitorImpl.java

```diff
@@ -0,0 +1,851 @@
+
+          if (!this.nodeThreads.isEmpty()) {
+            // stop node threads
+            for (Thread thread : this.nodeThreads.values()) {
+              thread.interrupt();
+            }
+            this.nodeThreads.clear();
+          }
+
+          final List<HostSpec> hosts = this.fetchTopologyAndUpdateCache(this.monitoringConnection.get());
+          if (hosts == null) {
+            // can't get topology
+            // let's switch to panic mode
+            Connection conn = this.monitoringConnection.get();
+            this.monitoringConnection.set(null);
+            this.isVerifiedWriterConnection = false;
+            this.closeConnection(conn);
+          }
+
+          if (this.highRefreshRateEndTimeNano > 0 && System.nanoTime() > this.highRefreshRateEndTimeNano) {
+            this.highRefreshRateEndTimeNano = 0;
+          }
+
+          // Do not log topology while in high refresh rate. It's noisy!
+          if (this.highRefreshRateEndTimeNano == 0) {
+            LOGGER.finest(Utils.logTopology(this.topologyMap.get(this.clusterId)));
+          }
+
+          this.delay(false);
+        }
+
+        if (this.ignoreNewFailoverRequestsEndTimeNano.get() > 0
+            && System.nanoTime() > this.ignoreNewFailoverRequestsEndTimeNano.get()) {
+          this.ignoreNewFailoverRequestsEndTimeNano.set(0);
+        }
+      }
+
+    } catch (final InterruptedException intEx) {
+      Thread.currentThread().interrupt();
+
+    } catch (final Exception ex) {
+      // this should not be reached; log and exit thread
+      if (LOGGER.isLoggable(Level.FINEST)) {
+        // We want to print full trace stack of the exception.
+        LOGGER.log(
+            Level.FINEST,
+            Messages.get(
+                "ClusterTopologyMonitorImpl.exceptionDuringMonitoringStop",
+                new Object[]{this.initialHostSpec.getHost()}),
+            ex);
+      }
+
+    } finally {
+      this.stop.set(true);
+
+      final Connection conn = this.monitoringConnection.get();
+      this.monitoringConnection.set(null);
+      this.closeConnection(conn);
+
+      LOGGER.finest(() -> Messages.get(
+          "ClusterTopologyMonitorImpl.stopMonitoringThread",
+          new Object[]{this.initialHostSpec.getHost()}));
+    }
+  }
+
+  protected boolean panicMode() {
+    return this.monitoringConnection.get() == null
+        || !this.isVerifiedWriterConnection;
+  }
+
+  protected Thread getNodeMonitoringThread(final HostSpec hostSpec, final @Nullable HostSpec writerHostSpec) {
+    final Thread thread = new NodeMonitoringThread(this, hostSpec, writerHostSpec);
+    thread.setDaemon(true);
+    return thread;
+  }
+
+  protected List<HostSpec> openAnyConnectionAndUpdateTopology() {
+
+    if (this.monitoringConnection.get() == null) {
+
+      Connection conn;
+
+      // open a new connection
+      try {
+        conn = this.pluginService.forceConnect(this.initialHostSpec, this.monitoringProperties);
```

Comment:
The method is called in a case when the cache has no topology. That means that the only know host to connect to is the initial host.  If the initial node is down at this time then there's no other option rather than to continue "being in panic" and try again and again.